webui/react/src/hooks/useSettings.ts

```diff
@@ -319,7 +317,7 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
       const url = mappedSettings ? `?${mappedSettings}` : '';
       navigate(url, { replace: true });
     });
-  }, [derivedOb, navigate, config, updateDB, initialLoading, loadableUser]);
+  }, [navigate, config, updateDB, initialLoading, loadableUser, settingsByKey]);
```

Comment:
You're creating a new `settingsByKey` every render so this just creates endless new subscriptions meaning an ever increasing number of API calls.